### PR TITLE
add periodic distance, second attempt

### PR DESCRIPTION
**Pull request by dkarrasch**
This is a clean implementation on current master, as (sort of) agreed on in #107. This is ready for review/merge.

### CODE REVIEW

**Comment by KristofferC** [comment]
Looks good to me. You verified the https://github.com/JuliaStats/Distances.jl/pull/107#issuecomment-473232505 benchmark I presume?

**Comment by dkarrasch** [comment]
Yes. I'm getting on my PR
```julia
julia> @btime colwise(dd, Xcol,Ycol)
  852.260 μs (2 allocations: 781.33 KiB)

julia> @btime colwise_naive(Xcol,Ycol)
  835.015 μs (2 allocations: 781.33 KiB)
```
and on the currently tagged version
```julia
julia> @btime colwise(dd, Xcol,Ycol)
  847.786 μs (2 allocations: 781.33 KiB)

julia> @btime colwise_naive(Xcol,Ycol)
  835.234 μs (2 allocations: 781.33 KiB)
```

File: README.md [hunk 1]
```diff
@@ -248,28 +252,30 @@ The table below compares the performance (measured in terms of average elapsed t
...
+| RenyiDivergence | 0.127884s |  0.114440s |  1.1175 |
+| JSDivergence | 0.114959s |  0.106575s |  1.0787 |
+| BhattacharyyaDist | 0.019642s |  0.010409s |  1.8870 |
+| HellingerDist | 0.019028s |  0.009799s |  1.9418 |
+| WeightedSqEuclidean | 0.011327s |  0.001252s |  9.0478 |
+| WeightedEuclidean | 0.011520s |  0.001338s |  8.6124 |
+| WeightedCityblock | 0.011308s |  0.002538s |  4.4547 |
+| WeightedMinkowski | 0.121861s |  0.105896s |  1.1508 |
+| WeightedHamming | 0.011249s |  0.002448s |  4.5945 |
+| SqMahalanobis | 0.189282s |  0.001452s | 130.3337 |
```
Comment (KristofferC) [inline README.md]: I'm a bit surprised by the relative slowdowns in the `pairwise` method here (and in some other places). Perhaps that is due to the CPU used here being weaker and not gaining as much from using BLAS.

Comment (dkarrasch) [reply README.md]: Hm, I overlooked that. Those BLAS-optimized `pairwise` methods are not touched by my PR or by the `parameter` trait, are they? Can we run the benchmark script on the original reference computer?

Comment (KristofferC) [reply README.md]: I'll try run it on my computer 9900k and see what I get.

Comment (dkarrasch) [reply README.md]: Donnu what went wrong. I'm getting much better values now, even better gains than were documented before in the README. I'll push the new tables in a minute.

**Pull request merged**